napari/util/colormaps/colormaps.py

```diff
@@ -1,7 +1,171 @@
+    dim : one of {1, 2, 3}
+        The dimensionality of the sequence.
+    n : int
+        How many points to generate.
+    seed : float or array of float, shape (dim,)
+        The seed from which to start the quasirandom sequence.
+
+    Returns
+    -------
+    pts : array of float, shape (n, dim)
+        The sampled points.
+
+    References
+    ----------
+    ..[1]: http://extremelearning.com.au/unreasonable-effectiveness-of-quasirandom-sequences/
+    """
+    phi1 = 1.6180339887498948482
+    phi2 = 1.32471795724474602596
+    phi3 = 1.22074408460575947536
+    seed = np.broadcast_to(seed, (1, dim))
+    phi = np.array([phi1, phi2, phi3])
+    g = 1 / phi
+    n = np.reshape(np.arange(n), (n, 1))
+    pts = (seed + (n * g[:dim])) % 1
+    return pts
+
+
+def _color_random(n, *, colorspace='lab', tolerance=0.0, seed=0.5):
+    """Generate n random RGB colors uniformly from LAB or LUV space.
+
+    Parameters
+    ----------
+    n : int
+        Number of colors to generate.
+    colorspace : str, one of {'lab', 'luv', 'rgb'}
+        The colorspace from which to get random colors.
+    tolerance : float
+        How much margin to allow for out-of-range RGB values (these are
+        clipped to be in-range).
+    seed : float or array of float, shape (3,)
+        Value from which to start the quasirandom sequence.
+
+    Returns
+    -------
+    rgb : array of float, shape (n, 3)
+        RGB colors chosen uniformly at random from given colorspace.
+    """
+    factor = 6  # about 1/5 of random LUV tuples are inside the space
+    expand_factor = 2
+    rgb = np.zeros((0, 3))
+    while len(rgb) < n:
+        random = _low_discrepancy(3, n * factor, seed=seed)
+        if colorspace == 'luv':
+            raw_rgb = colorconv.luv2rgb(random * LUVRNG + LUVMIN)
+        elif colorspace == 'rgb':
+            raw_rgb = random
+        else:  # 'lab' by default
+            raw_rgb = colorconv.lab2rgb(random * LABRNG + LABMIN)
+        rgb = _validate_rgb(raw_rgb, tolerance=tolerance)
+        factor *= expand_factor
+    return rgb[:n]
+
+
+def label_colormap(labels, seed=0.5):
+    """Produce a colormap suitable for use with a given label set.
+
+    Parameters
+    ----------
+    labels : array of int
+        A set of labels or label image.
+    seed : float or array of float, length 3
+        The seed for the low discrepancy sequence generator.
+
+    Returns
+    -------
+    cmap : vispy.color.Colormap
+        A colormap for use with ``labels``. The labels are remapped so that
+        the maximum label falls on 1.0, since vispy requires colormaps to map
+        within [0, 1].
+
+    Notes
+    -----
+    0 always maps to fully transparent.
+    """
+    unique_labels = np.unique(labels)
+    n = len(unique_labels)
+    max_label = np.max(unique_labels)
+    unique_labels_float = unique_labels / max_label
+    midpoints = np.convolve(unique_labels_float, [0.5, 0.5], mode='valid')
+    eps = np.finfo(midpoints.dtype).eps
```

Comment:
It looks like this doesn't get used. What is it for?

Reply:
A relic from another age. =P Removed.